DirectProgramming/DPC++FPGA/ReferenceDesigns/decompress/README.md

```diff
@@ -0,0 +1,427 @@
+       ```
+
+3. (Optional) As the above hardware compile may take several hours to complete, FPGA precompiled binaries (compatible with Linux* Ubuntu* 18.04) can be downloaded <a href="https://iotdk.intel.com/fpga-precompiled-binaries/latest/decompress.fpga.tar.gz" download>here</a>.
+
+### On a Windows* System
+1. Generate the `Makefile` by running `cmake`.
+     ```
+   mkdir build
+   cd build
+   ```
+   To compile for the Intel® PAC with Intel Arria® 10 GX FPGA, run `cmake` using the command:
+    ```
+    cmake -G "NMake Makefiles" ..
+   ```
+   Alternatively, to compile for the Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX), run `cmake` using the command:
+   ```
+   cmake -G "NMake Makefiles" .. -DFPGA_BOARD=intel_s10sx_pac:pac_s10
+   ```
+
+2. Compile the design through the generated `Makefile`. The following build targets are provided, matching the recommended development flow:
+   * Compile for emulation (fast compile time, targets emulated FPGA device):
+     ```
+     nmake fpga_emu
+     ```
+   * Generate the optimization report:
+     ```
+     nmake report
+     ``` 
+   * An FPGA hardware target is not provided on Windows*. 
+
+*Note:* The Intel® PAC with Intel Arria® 10 GX FPGA and Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX) do not yet support Windows*. Compiling to FPGA hardware on Windows* requires a third-party or custom Board Support Package (BSP) with Windows* support.<br>
+*Note:* If you encounter any issues with long paths when compiling under Windows*, you may have to create your ‘build’ directory in a shorter path, for example c:\samples\build.  You can then run cmake from that directory, and provide cmake with the full path to your sample directory.
+
+## Troubleshooting
+
+If an error occurs, you can get more details by running `make` with
+the `VERBOSE=1` argument:
+``make VERBOSE=1``
+For more comprehensive troubleshooting, use the Diagnostics Utility for
+Intel® oneAPI Toolkits, which provides system checks to find missing
+dependencies and permissions errors.
+[Learn more](https://software.intel.com/content/www/us/en/develop/documentation/diagnostic-utility-user-guide/top.html).
+
+### In Third-Party Integrated Development Environments (IDEs)
+
+You can compile and run this Reference Design in the Eclipse* IDE (in Linux*) and the Visual Studio* IDE (in Windows*). For instructions, refer to the following link: [Intel® oneAPI DPC++ FPGA Workflows on Third-Party IDEs](https://software.intel.com/en-us/articles/intel-oneapi-dpcpp-fpga-workflow-on-ide)
+
+## Running the Reference Design
+
+ 1. Run the sample on the FPGA emulator (the kernel executes on the CPU).
+     ```
+     ./decompress.fpga_emu    (Linux)
+     decompress.fpga_emu.exe  (Windows)
+     ```
+
+2. Run the sample on the FPGA device.
+     ```
+     ./decompress.fpga        (Linux)
+     ```
+
+### Example of Output
+You should see output similar to the following in the console for the GZIP and SNAPPY versions, respectively:
```

Comment:
How is the user selecting between a GZIP and SNAPPY decompression? Is it chosen at compile time? or are both versions compiled to HW? Does the exe figures out if it needs to decompress using GZIP or SNAPPY?

Reply:
Using a compile-time macro. From main.cpp:
```cpp
// ensure only one of GZIP and SNAPPY is defined
#if defined(GZIP) and defined(SNAPPY)
static_assert(false, "Only one of GZIP and SNAPPY can be defined!");
#endif

// if neither of GZIP and SNAPPY is defined, default to SNAPPY
#if not defined(GZIP) and not defined(SNAPPY)
#define SNAPPY
#endif
```

I will add a not about this in the README

Reply:
I do have this. But I included this in case someone goes rogue and doesn't use CMake.